src/main/resources/validationrules/iac/iac-case-role-validation-admin.drl

```diff
@@ -0,0 +1,106 @@
+    $ra.setAttribute("caseType", $c.getCaseTypeId());
+    $ra.setStatus(Status.CREATE_APPROVED);
+    $ra.log("Stage 1 approved : create_specific_access_requested_admin_case_role");
+    update($ra);
+    logMsg("Rule : create_specific_access_requested_admin_case_role");
+end;
+
+/*
+ * IAC "specific-access-admin" case creation and case-allocator as a assigner
+ */
+rule "case_allocator_create_specific_access_admin_case_role"
+when
+    $f : FeatureFlag(status && flagName == FeatureFlagEnum.IAC_SPECIFIC_1_0.getValue())
+    $rq: Request(
+             // The assigner must be the current user.
+             assignerId == authenticatedUserId,
+             process == "specific-access",
+             reference != null)
+    $ra: RoleAssignment(
+             status == Status.CREATE_REQUESTED,
+             roleType == RoleType.CASE,
+             roleName == "specific-access-admin",
+             roleCategory == RoleCategory.ADMIN,
+             grantType == GrantType.BASIC,
+             notes != null && notes.size > 0 && notes.toString().length() > 5, /* Make sure there is at least one note with content in it */
+             attributes["caseId"] != null && $caseId : attributes["caseId"].asText(),
+             attributes["jurisdiction"] != null && attributes["jurisdiction"].asText() in ("IA"))
+    $c:  Case(
+             id == $caseId)
+    $e: ExistingRoleAssignment(
+             actorId == $rq.assignerId,
+             roleType == RoleType.ORGANISATION,
+             roleName == "case-allocator",
+             grantType == GrantType.STANDARD,
+             classification.isAtLeast($ra.classification),
+             endTime == null || endTime.isAfter(ZonedDateTime.now()),
+             beginTime == null || !beginTime.isAfter(ZonedDateTime.now()),
+             attributes["jurisdiction"] != null && attributes["jurisdiction"].asText() == $c.jurisdiction,
+             attributes["caseType"] == null || attributes["caseType"].asText() == $c.caseTypeId,
+             attributes["region"] == null || attributes["region"].asText() != $c.getRegion(),
+             attributes["baseLocation"] == null || attributes["baseLocation"].asText() != $c.getBaseLocation(),
+             attributes["managedRoleCategory"] == null || attributes["managedRoleCategory"].asText() == $ra.roleCategory.toString(),
+             attributes["managedRole"] == null || attributes["managedRole"].asText() == $ra.roleName)
+then
+    // We copy the authorisations to allow the case config to require an authorisation
+    // for (particularly judicial) role mappings
+    $ra.setAuthorisations($e.getAuthorisations());
+    $ra.setAttribute("jurisdiction", $c.getJurisdiction());
+    $ra.setAttribute("caseType", $c.getCaseTypeId());
+    $ra.setStatus(Status.CREATE_APPROVED);
+    $ra.log("Stage 1 approved : case_allocator_create_specific_access_admin_case_role");
+    update($ra);
+    logMsg("Rule : case_allocator_create_specific_access_admin_case_role");
+end;
```

Comment:
The delete rule is missing..I think we need to add it in the separate individual file or it would be moved in the common core folder..for now move it to the individual file which is easier to manage. you can add a condition of releCategory in that.